docs/softwaredevelopment.md

```diff
@@ -1 +1,124 @@
-# Software development
\ No newline at end of file
+# Software development
+
+## How to name software products
+Read this guide when you need to name components for your software product, for example applications, software libraries, plugins or frameworks. Your users should understand what something does from its name.
+
+### User needs
```

Comment:
I don't think we need this small paragraph – it seems to be covered in the next one?